src/ui/store/reducers/editHistory.ts

```diff
@@ -0,0 +1,130 @@
+import {AnyAction} from 'redux';
+import {Delta as JDPDelta} from 'jsondiffpatch';
+
+import {
+    INIT_EDIT_HISTORY_UNIT,
+    RESET_EDIT_HISTORY_UNIT,
+    ADD_EDIT_HISTORY_POINT,
+    SET_EDIT_HISTORY_POINT_INDEX,
+    EditHistoryAction,
+    SET_EDIT_HISTORY_CURRENT_STATE,
+} from '../actions/editHistory';
+
+export type Diff = JDPDelta;
+
+export type EditHistoryUnit = {
+    diffs: (Diff | null)[];
+    pointIndex: number;
+    pointState: unknown;
+    setState: ({state}: {state: unknown}) => AnyAction;
```

Comment:
Does this mean that we store the way to apply history changes in the store?

Reply:
Yes